pallets/foreign-investments/src/impls/invest.rs

```diff
@@ -1117,3 +1117,348 @@ where
 		}
 	}
 }
+
+#[cfg(test)]
+mod tests {
+	use cfg_traits::IdentityCurrencyConversion;
+	use frame_support::{assert_err, assert_ok};
+	use rand::{rngs::StdRng, seq::SliceRandom, SeedableRng};
+
+	use super::*;
+
+	#[derive(Clone, Copy, PartialEq, Debug)]
+	enum CurrencyId {
+		Foreign,
```

Comment:
`nit` I understand why we call this "Foreign" but the idea that we want this to represent AFAIK is that it should be a stable coin. Not all Foreign assets will be stable coins / pool currencies. So I'd say the code and the mental model one would build from it would be more sensible to name this `StableCoin`.

Reply:
But the pool's currency is also a stable coin, right? I mean, both would be stable coins, the foreign and the pool.

Reply:
Wanted to know the opinion of @wischli because I think this was an open topic. Right now, I've chosen those names because everything in the implementation is called as them.

Reply:
Yeah you got a good point! A pool currency is pegged to a stable coin so by definition it is indeed a stable coin afaik. All good for me 👍